tests/unit/test_task_events.py

```diff
@@ -0,0 +1,89 @@
+from typing import Dict
+from unittest.mock import Mock
+
+import pytest
+from pytest_mock import MockerFixture
+
+from nidaqmx import Task
+from nidaqmx.task import _TaskEventType
+from tests.unit._grpc_utils import create_grpc_options
+from tests.unit._task_utils import (
+    expect_create_task,
+    expect_get_task_name,
+    register_event_handler,
+    register_event_handlers,
+)
+
+
+@pytest.mark.parametrize("event_type", _TaskEventType)
+def test___grpc_event_registered___leak_task___raises_resource_warning(
+    event_type: _TaskEventType, interpreter: Mock, mocker: MockerFixture
+):
+    expect_create_task(interpreter, "GrpcTaskHandle")
+    expect_get_task_name(interpreter, "GrpcTask")
+    grpc_options = create_grpc_options(mocker)
+    task = Task("GrpcTask", grpc_options=grpc_options)
+    register_event_handler(mocker, task, event_type)
+
+    with pytest.warns(ResourceWarning, match="Event handlers may still be active."):
+        task.__del__()
+
+    task.close()
+
+
+def test___events_registered_and_clear_task_error_raised___close___task_resources_cleaned_up_and_clear_task_error_raised(
+    interpreter: Mock, mocker: MockerFixture
+):
+    expect_create_task(interpreter, "MyTaskHandle")
+    expect_get_task_name(interpreter, "MyTask")
+    task = Task("MyTask")
+    event_handlers = register_event_handlers(mocker, task, _TaskEventType)
+    interpreter.clear_task.side_effect = RuntimeError("clear_task")
```

Comment:
Ahh, I was wondering what the side effect was going to be for. I like it!

Reply:
Are you referring to the side effect field in `BaseEventObserver`? This side effect field is in the `Mock` class: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock

Now that I think about it, passing an exception for the side effect is really what I wanted in `BaseEventObserver`.